### PR TITLE
Bloom filters

### PR DESCRIPTION
**Pull request by Krukov**

### CODE REVIEW

File: cashews/utils/bitarray.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class Bitarray:
+    __slots__ = ("_value",)
+
+    def __init__(self, value, base=10):
+        self._value = int(value, base)
+
+    def get(self, index: int, size: int = 1) -> int:
```
Comment (unmade) [inline cashews/utils/bitarray.py:7]: Seems like this method can re-written using bit-wise operator. 

Basically this method will do almost the same:
```python
def get(value, idx, size):
    return value << size & (idx + 1)
```

Comment (unmade) [inline cashews/utils/bitarray.py:7]: It seems like the whole method can be done using bitwise operations. It should be faster and easier to understand

```python
def get(value, idx, size):
    res = 0
    for i in range(idx, idx + size):
        res <<= 1
        if value & (2 ** i):
            res |= 1
    return res
```

File: tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    assert b.get(3) == 0
+    assert b.get(4) == 0
+    assert b.get(5) == 1
+    assert b.get(6) == 0
+    assert b.get(7) == 0
+
+
+def test_bitarray_get_size_2():
+    b = Bitarray("100010", 2)
+    assert b.get(0, 2) == 2
```
Comment (unmade) [inline tests/test_utils.py:18]: it should equal to `1`, since we sliced from the end and took only two bits (in that case - `0` and `1`)

**Review by unmade — CHANGES_REQUESTED** [review]

File: cashews/utils/split_hash.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def get_hashes(key: str, k: int, max_i: int):
+    """
+    return array with bit indexes for given value (key) [23, 45, 15]
+    """
+    assert max_i >= k
+    str_hash = str(_get_string_int_hash(key))
+    indexes = set()
+    for _hash in _split_string_for_chunks(str_hash, k):
```
Comment (Krukov) [inline cashews/utils/split_hash.py:11]: remake with bit operations

**Pull request merged**